File: 5/python/5.py
#----------------------------------------------------------------------------------------------
# 2520 is the smallest number that can be divided by each of the numbers 
# from 1 to 10 without any remainder. What is the smallest positive number
# that is evenly divisible by all of the numbers from 1 to 20?
#----------------------------------------------------------------------------------------------
texte = 1 
x = 1 
soma = 0 

while(texte == 1):
    for i in range(1,21):
        if(x%i == 0 ):
            soma = soma +1 
    if(soma == 20 ):    
        texte = 5 
    else:
        soma = 0 
        x = x + 1 

print(x)
     

File: 5/fortran/5.f90
!------------------------------------------------------------------------------
! 2520 is the smallest number that can be divided by each of the numbers 
! from 1 to 10 without any remainder. What is the smallest positive number
! that is evenly divisible by all of the numbers from 1 to 20?
!------------------------------------------------------------------------------
program projecteuler5
implicit none 

integer     :: i, j, num, texte, soma
texte = 1 
soma = 0 
i = 1 

DO WHILE(texte == 1)
    do j = 1, 20
        if(mod(i, j)==0)then
            soma = soma +1 
        endif
    enddo 
        if(soma == 20 )then 
            texte = 5 
        endif 
        num = i 
        i = i+1 
        soma = 0
ENDDO

write(*,*)num
end program projecteuler5